dotnet/Examples/GcpAuth.cs

```diff
@@ -0,0 +1,113 @@
+        ///     	type="iam" \
+        ///     	policies="dev-policy" \
+        ///     	bound_service_accounts="my-service@my-project.iam.gserviceaccount.com"
+        /// Your Vault instance must also be configued with GCP credentials to perform API calls to IAM, like so:
+        ///     vault write auth/gcp/config credentials=@path/to/server/creds.json
+        /// Learn more at https://www.vaultproject.io/docs/auth/gcp
+        /// </summary>
+        public string GetSecretGcp()
+        {
+            var vaultAddr = Environment.GetEnvironmentVariable("VAULT_ADDR");
+            var roleName = Environment.GetEnvironmentVariable("GCP_ROLE");
+
+            // Learn about authenticating to GCS with service account credentials at https://cloud.google.com/docs/authentication/production
+            if(String.IsNullOrEmpty(Environment.GetEnvironmentVariable("GOOGLE_APPLICATION_CREDENTIALS")))
+            {
+                Console.WriteLine("WARNING: Environment variable GOOGLE_APPLICATION_CREDENTIALS was not set. IAM client for JWT signing will fall back to default instance credentials.");
+            }
+
+            var jwt = SignJWT();
+ 
+            IAuthMethodInfo authMethod = new GoogleCloudAuthMethodInfo(roleName, jwt);
+            var vaultClientSettings = new VaultClientSettings(vaultAddr, authMethod);
+
+            IVaultClient vaultClient = new VaultClient(vaultClientSettings); 
+
+            // We can retreive the secret after creaing our VaultClient object
+            Secret<SecretData> kv2Secret = null;
+            try
+            {   
+                kv2Secret = vaultClient.V1.Secrets.KeyValue.V2.ReadSecretAsync(path: "/creds").Result;
+            }
+            catch(Exception e)
+            {
+                Console.WriteLine($"An error occurred while retreiving secret: ", e.Message); 
+                return string.Empty;
+            }
+            
+            var password = kv2Secret.Data.Data["password"];
+            
+            return password.ToString();
+        }
+
+        /// <summary>
+        /// Generate signed JWT from GCP IAM
+        /// </summary>
+        private string SignJWT()
+        {
+            var roleName = Environment.GetEnvironmentVariable("GCP_ROLE");
+            var svcAcctName = Environment.GetEnvironmentVariable("GCP_SERVICE_ACCOUNT_NAME");
+            var gcpProjName = Environment.GetEnvironmentVariable("GOOGLE_CLOUD_PROJECT");
+
+            IamService iamService = new IamService(new BaseClientService.Initializer
+            {
+                HttpClientInitializer = GetCredential(),
+                ApplicationName = "Google-iamSample/0.1",
+            });
+
+            string svcEmail = $"{svcAcctName}@{gcpProjName}.iam.gserviceaccount.com";
+            string name = $"projects/-/serviceAccounts/{svcEmail}";  
+
+            TimeSpan currentTime = (DateTime.UtcNow - new DateTime(1970, 1, 1));
```

Comment:
i assume it works, but it's not clear why you need to remove the time. epoch?

Reply:
I can add a comment to make this more clear, but it's essentially the time format that the Google Api requires the expiration to be in